examples/providerconfig/v1beta1/pod-identity.yaml

```diff
@@ -0,0 +1,60 @@
+# SPDX-FileCopyrightText: 2024 The Crossplane Authors <https://crossplane.io>
+#
+# SPDX-License-Identifier: CC0-1.0
+
+apiVersion: aws.upbound.io/v1beta1
+kind: ProviderConfig
+metadata:
+  name: pod-identity
+spec:
+  credentials:
+    source: WebIdentity
+    webIdentity:
+      roleARN: arn:aws:iam::123456789012:role/pod-identity-role
+      tokenConfig:
+        fs:
+          path: /var/run/secrets/pods.eks.amazonaws.com/serviceaccount/eks-pod-identity-token
```

Comment:
maybe this is a future improvement in crossplane but it would be nice if we could have this be easier to implement without having to hard code path 🤔

Reply:
I'm surprised that it's necessary to set this explicitly, and that the SDK doesn't "Just Work" when instantiated in a pod that's assigned a PodIdentity. Maybe what we really need is to add an explicit `PodIdentity` to the list of supported auth types?

Reply:
today we need to set it explicit - because our GO SDK is not up2date with this feature

Reply:
Is there an issue to track that?

Reply:
Quick question, if not a clarification:

IRSAv1 had a huge pain point in needing to include the roleARN that you're going to annotate service accounts with, so that the OIDC trust loop worked.

With IRSAv2 (Pod Identity Controller) I can configure all the AWS IAM/EKS components out of cluster, and as long as a pod starts up in the nominated namespace, and with the service account I picked, it should inherit the permissions at start via the injected token provided by the pod access controller.

Given this; if I have to mangle my helm to be coupled to the IAM ARN, I'm not sure I'm saving any hassle here - I want to just add a pod identity out of band (outside EKS, not involving Crossplane, since the host cluster is managed with terraform) and have Crossplane start up and use the IAM permissions granted to it.

Am I misunderstanding things here?

It's a relatively new concept but I've had success with this decoupled model with External Secrets controller for example.

Additionally as pointed out by @mbbush if we need to configure this, rather than letting it fall through the SDK default authentication methods a lot of the benefit of "it just works" that; the pod identity controller, the associated admission mutation webhook and the tokens just being in the expected place... start to lose some value.